### DISCUSSION TITLE
Beeline indication in gpx file possible?

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#395] [Q&A] [answered]
We plan to ride the ACT Portugal next year. Parts of the provided track aren't even defined in OSM - so I've to use "beeline" for this steps. That's not really a problem, I checked the tracks in Google Earth and can take the original track as overlay to follow. Btw: I'm using the Mountainbike profile for this.

Because you can't convince all people to get rid of their Garmin stuff, I try to plan totally in .gpx format. But then the information "beeline" is lost after exporting a gpx file with track and route. In navigation format the -src-beeline-/src- parameter is kept, but w/o the route for sure. Is there a chance to get this parameter in "normal" gpx files as well?

Here the tests: [beeline.zip](https://github.com/devemux86/cruiser/files/13636869/beeline.zip)

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for the suggestion.

How do you mean it in normal gpx files? In the rtept?
And how it can be used in other devices that do not know the beeline?

**Reply by WolfgangZ42** [reply]
Yes in the RTEPT. I don't know if Garmin supports "beeline" - but I'd like that Cruiser just remembers the straightline state of waypoints after export/import. (As it works with .kurviger files.)
I'll change wording from "beeline" to "straightline" in the future. More clear and compatible with the Cruiser description - "Beeline" is even a Company-Name, I learned via Google today.

**Reply by devemux86** [reply]
This is the purpose of the navigation track, it contains the route with all data,
turn instructions, waypoints, weights, etc. And it can be used **offline**.
- https://github.com/devemux86/cruiser/discussions/250

Normal GPX require route calculation during import.
So it seems the waypoint types already exist in normal GPX.
The suggestion is to also include the weight to complete the route requests?

**Reply by WolfgangZ42** [reply]
> The suggestion is to also include the weight to complete the route requests?

exactly

> This is the purpose of the navigation track, it contains the route with all data,
turn instructions, waypoints, weights, etc.

The navigation track doesn't contain waypoints anymore, as far as I understand - see my examples.

**Reply by devemux86** [reply]
> The navigation track doesn't contain waypoints anymore, as far as I understand - see my examples.

Navigation tracks contain everything or else we could not read offline the routes.

The waypoints are described with the tags:
```xml
<sym>pass_place</sym>
<type>via</type>
<type>shaping</type>
```

**Reply by WolfgangZ42** [reply]
That's true, but it is a TRKPT not a RTEPT. 
![image](https://github.com/devemux86/cruiser/assets/121806238/735ca9d2-1cfb-4f60-92e4-1be10defcb5a)

It wouldn't a problem if I would be on road (or better: offroad) alone or only with Cruiser users. I'm trying to support my Garmin fellows with the best possible route. And I doubt that the navigation track works for a BMW Navigator VI. But I don't know much about Garmin, obviously.

**Reply by devemux86** [reply]
Indeed everything is included in the `trk`. We don't currently use the `wpt` or `rtept`.

---

Navigation (offline) tracks are currently supported by:
- BRouter
- Cruiser
- Locus
- [RouteYou](https://www.routeyou.com/)

**Reply by WolfgangZ42** [reply]
Obviously I didn't consider that all information is available again after importing a navigation gpx file (via points, shaping points, straight lines). In this respect, I can use the navigation format for my intermediate versions of planning and make the final version available to the Garmin people as a normal gpx file. Whether they can then use straight lines remains to be seen. 
Thank you for your patience!

**Comment by devemux86** [comment]
@0709wiwiwi in navigation GPX the weight is stored in  `src` of `trkpt`.

We could also store the weight in the `src` field of the `wpt` or `rtept`?
It is currently unused (the `sym` and `type` are used for waypoint types).

- The above is during the export.
- In import of normal GPX Cruiser should use the found beelines
with supported routers, like BRouter?

Is this the suggestion?

**Reply by devemux86** [reply]
Let me see what can be done. :slightly_smiling_face:

**Reply by 0709wiwiwi** [reply]
Does Garmin indicate manual sections in gpx (rte) routes ? 
No idea. A Basecamp gpx rte export is attached. 
[BASECAMP RTE.gpx.txt](https://github.com/devemux86/cruiser/files/13639175/BASECAMP.RTE.gpx.txt)

Anyway in a gpx navigation track is indicated.
In the Cruiser gpx wpt or rte exports not.
It would also **only be useful** for Cruiser.
A manual gpx navigation track is attached. 
Without beeline clock guiding in the sym ;-)
[Cruiser Beeline NAVTRK.gpx.txt](https://github.com/devemux86/cruiser/files/13640778/Cruiser.Beeline.NAVTRK.gpx.txt)

**Reply by WolfgangZ42** [reply]
![image](https://github.com/devemux86/cruiser/assets/121806238/250c0010-fd6b-49e4-bd24-9d1b8a4a99e4)
👍😉

**Reply by 0709wiwiwi** [reply]
_The CRUISER navigation track shown is a "toyish" design but interesting..
For those who like to do some extra wintertest see the next.
The test files are the zip._
[CR.zip](https://github.com/devemux86/cruiser/files/13651219/CR.zip)

A. Cruiser is quite unique as it announces the beeline track sections.
But multiple TTS "straight line" commands are sure not informative .
Announcing the guiding direction to follow is desirable and useful.

B. In the display notice that the characters are connected by a visual track line.
It would be more nice and clear if no connection was visible between them.

Can it be done by a gpx file import ?
To test Import the gpx tracks as visible template only !

A short CR gpx navtrack.
File gpx 1. (Cruiser export)
Contains 1 gpx file with 1 (trk)track and 1 (trkseg)tracksegment
Between C and R a visible track line is shown.

File 2.
Contains 1 gpx file with 2 (trk) tracks, each has 1 (trkseg) tracksegment.
This navigation track can only be navigated per selected single navtrack C or R.
Between C and R no visible line is shown.

File gpx 3.
Contains 1 gpx file with 1 (trk) track and 2 (trkseg)track segments.
Between C and R the track line is transparant. (Not all apps do support this)

No worry !
Cruiser does NOT generate gpx navigation tracks with (trkseg) tracksegments.
Cruiser can add (trkseg)tracksegments in the exported gpx trackrecord files.

**Comment by devemux86** [comment]
Straight lines in normal GPX will be available in the next Cruiser.

This is the new normal GPX export of the above example (with straight lines + shaping points):
[beeline2.gpx.txt](https://github.com/devemux86/cruiser/files/13674674/beeline2.gpx.txt)

---

Info: Besides BRouter, OpenRouteService also supports straight lines in Cruiser.

**Reply by devemux86** [reply]
Straight lines are supported in both Export + Import of normal GPX in the latest:
- https://github.com/devemux86/cruiser/discussions/390

@WolfgangZ42 please test your GPX and let me know. :slightly_smiling_face:

**Reply by WolfgangZ42** [reply]
Works as expected, as always! 👍😄
As I said, with Cruiser I (and two other participants) can follow the whole ACT-Portugal track exactly (5 days, ~1500 km, incl. 4 short beeline parts) - but I want to give the others a .gpx-file to feed their Garmin-Navi with, without just turning on "follow the track".
I am about to install BaseCamp for testing ... 🥴

**Comment by devemux86** [comment, marked as answer]
Implemented in [Cruiser 3.0.37](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
